kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala

```diff
@@ -102,6 +102,25 @@ class SparkProcessBuilder(
     }
   }
 
+  private lazy val sparkCoreScalaVersion: String = {
+    Paths.get(sparkHome, "jars").toFile
+      .list((_, name) => name.matches("^spark-core_.*\\.jar$"))
+      .map { p => p.substring(p.indexOf("_") + 1, p.lastIndexOf("-")) }
```

Comment:
does it work with vendor versions like 

- `spark-core_2.13-3.5.0-abc-20230921.jar`
- `spark-core_2.13-3.5.0-xyz-1.2.3.jar`
- `spark-core_2.13-3.5.0.1.jar`

Reply:
I think the simplest and most robust way is to use regex to extract the Scala binary version directly

Reply:
Changed to use regex for extraction.